pep-0667.rst

```diff
@@ -0,0 +1,212 @@
+Type: Standards Track
+Content-Type: text/x-rst
+Created: 30-Mar-2021
+Post-History: 
+
+
+Abstract
+========
+
+This document describes a PEP 517 style method for the installation of packages
+in editable mode.
+
+Motivation
+==========
+
+Python programmers want to be able to develop packages without having to
+install (i.e. copy) them into ``site-packages``, for example, by working in a
+checkout of the source repository.
+
+While this can be done by adding the relevant source directories to
+``PYTHONPATH``, ``setuptools`` provides the ``setup.py develop`` mechanism that
+makes the process easier, and also installs dependencies and entry points such
+as console scripts. ``pip`` exposes this mechanism via its ``pip install
+--editable`` option.
+
+The installation of projects in such a way that the python code being
+imported remains in the source directory is known as the *editable*
+installation mode.
+
+Now that PEP 517 provides a mechanism to create alternatives to setuptools, and
+decouple installation front ends from build backends, we need a new mechanism
+to install packages in editable mode.
+
+Rationale
+=========
+
+PEP 517 deferred "Editable installs", meaning non-``setup.py``
+distributions lacked that feature. The only way to retain ``editable`` installs
+for these distributions was to provide a compatible ``setup.py develop``
+implementation. By definining an editable hook other build frontends gain
+parity with ``setup.py``.
+
+Terminology and goals
+=====================
+
+The editable installation mode implies that the source code of the project
+being installed is available in a local directory.
+
+Once the project is installed in editable mode, users expect that changes to
+the project *python* code in the local source tree become effective without the
+need of a new installation step.
+
+Some kind of changes, such as the addition or modification of entry points, or
+the addition of new dependencies, require a new installation step to become
+effective. These changes are typically made in build backend configuration
+files (such as ``pyproject.toml``), so it is consistent with the general user
+expectation that *python* source code is imported from the source tree.
+
+The modification of non-python source code such a C extension modules obviously
+require a compilation and/or installation step to become effective. The exact
+steps to perform will remain specific to the build backend used.
+
+When a project is installed in editable mode, users expect the installation to
+behave identically as a regular installation. Depending on the way build
+backend implement this specification, some minor differences may be visible
+such as the presence of additional files that are in the source tree and would
+not be part of a regular install. Build backends are encouraged to document
+such potential differences.
+
+The Mechanism
+=============
+
+This PEP adds a single optional hook to the PEP-517 backend interface. The hook
```

Comment:
Other than here and in the title you've written PEP 517 (i.e. without the hyphen)